.github/workflows/scheduled-ansible-pull.yml

```diff
@@ -0,0 +1,58 @@
+name: Check if the collection is up to date with the ansible role
+
+on:
+  - workflow_dispatch
+  - schedule:
+    - cron: '0 4 * * *'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    outputs:
+      check-output: ${{ steps.check-diff.outputs.sync-needed }}
+    steps:
+      - name: Check diff between roles/agent and ansible-role main branch
+        id: check-diff
+        run: |
+          mkdir /tmp/ansible-role
+          git clone git@github.com:DataDog/ansible-datadog.git /tmp/ansible-role
+          ERROR_CODE=0
+          diff -x "\.*" -x CHANGELOG.md -x CONTRIBUTING.md -x LICENSE -x ci_test -x manual_tests -x "requirements*" -pur roles/agent/ /tmp/ansible-role || ERROR_CODE=$?
+          if [ $ERROR_CODE -ne 0 ]; then
+            echo "sync-needed=true" >> $GITHUB_OUTPUT
+          fi
+
+  execute:
+    runs-on: ubuntu-latest
+    needs: check
+    if: "${{ needs.check.outputs.sync-needed == 'true' }}"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@master
+        with:
+          repository: DataDog/ansible-datadog
+          path: roles/agent
+      - name: Cleanup checked out repository
+        run: |
+          find roles/agent/ -not -path '*defaults*' -not -path '*tasks*' -not -path '*templates*' -not -path '*handlers*' -not -name '.gitkeep' -not -path '*meta*' -not -name 'README.md' -type f -exec rm -r {} \;
+          touch roles/agent/.gitkeep
```

Comment:
Does this make a difference? The file is not `git add`ed so I'd expect it to be gone by the end of the workflow